homeassistant/components/blebox/light.py

```diff
@@ -166,7 +167,7 @@ async def async_turn_on(self, **kwargs):
                 f"Turning on '{self.name}' failed: Bad value {value}"
             ) from exc
 
-        if effect is not None:
+        if effect is not None and self.effect_list is not None:
```

Comment:
`self.effect_list.index(effect)` is invalid if `self.effect_list` is `None`.

Reply:
It seems it can indeed be None based on a quick review of `blebox_uniapi`

It means this should be made conditional in the initializer:
```diff
-        self._attr_supported_features = LightEntityFeature.EFFECT
+        if self.effect_list:
+            self._attr_supported_features = LightEntityFeature.EFFECT
```

Reply:
Actually it returns an empty list in blebox_uniapi: https://github.com/blebox/blebox_uniapi/blob/a404d61883ad6b93b6833286f66149f0cdf06c81/blebox_uniapi/light.py#L242-L247
```python
    @property
    def effect_list(self):
        if isinstance(self._effect_list, dict):
            return list(self._effect_list.values())
        else:
            return []
```

But I guess that it still applies...

Reply:
Nice, I didn't find that 👍
Yeah, if the list is empty there's no need to flag support for effects.

Reply:
I've reworked the changes, and improved the type hints a little more.
It means this change is no longer required, as `self.effect_list` cannot be `None`.